vms/platformvm/state/diff_test.go

```diff
@@ -112,6 +113,154 @@ func TestDiffCurrentSupply(t *testing.T) {
 	assertChainsEqual(t, state, d)
 }
 
+func TestDiffExpiry(t *testing.T) {
+	type op struct {
+		put   bool
+		entry ExpiryEntry
+	}
+	tests := []struct {
+		name             string
+		initialExpiries  []ExpiryEntry
+		ops              []op
+		expectedExpiries []ExpiryEntry
+	}{
+		{
+			name: "empty noop",
+		},
+		{
+			name: "insert",
+			ops: []op{
+				{
+					put:   true,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+			},
+			expectedExpiries: []ExpiryEntry{
+				{Timestamp: 1},
+			},
+		},
+		{
+			name: "remove",
+			initialExpiries: []ExpiryEntry{
+				{Timestamp: 1},
+			},
+			ops: []op{
+				{
+					put:   false,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+			},
+		},
+		{
+			name: "add and immediately remove",
+			ops: []op{
+				{
+					put:   true,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+				{
+					put:   false,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+			},
+		},
+		{
+			name: "add + remove + add",
+			ops: []op{
+				{
+					put:   true,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+				{
+					put:   false,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+				{
+					put:   true,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+			},
+			expectedExpiries: []ExpiryEntry{
+				{Timestamp: 1},
+			},
+		},
+		{
+			name: "everything",
+			initialExpiries: []ExpiryEntry{
+				{Timestamp: 1},
+				{Timestamp: 2},
+				{Timestamp: 3},
+			},
+			ops: []op{
+				{
+					put:   false,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+				{
+					put:   false,
+					entry: ExpiryEntry{Timestamp: 2},
+				},
+				{
+					put:   true,
+					entry: ExpiryEntry{Timestamp: 1},
+				},
+			},
+			expectedExpiries: []ExpiryEntry{
+				{Timestamp: 1},
+				{Timestamp: 3},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		require := require.New(t)
+
+		state := newTestState(t, memdb.New())
+		for _, expiry := range test.initialExpiries {
+			state.PutExpiry(expiry)
+		}
+
+		d, err := NewDiffOn(state)
+		require.NoError(err)
+
+		otherExpiries := set.Of(test.initialExpiries...)
+		for _, op := range test.ops {
+			if op.put {
+				d.PutExpiry(op.entry)
+			} else {
+				d.DeleteExpiry(op.entry)
+			}
+			otherExpiries.Add(op.entry)
+		}
+		otherExpiries.Remove(test.expectedExpiries...)
```

Comment:
What is other expiries doing here? This test flow seems a bit odd to me

Reply:
It is the expiries that shouldn't be in state

Reply:
Could you add a comment explaining this into the code? The test setup of:

1. add the initial expiries
2. add all of the expiry ops (both put/remove)
3. remove all of the expected values

does not seem clear just from the variable names that this results in the expiries not in state

Reply:
1. I changed the name from `otherExpiries` to `unexpectedExpiries`.
2. I removed `expectedExpiries` from the test struct and just calculated them explicitly in the test body.